Bottom: e4408e3ac9afe19fbfcc20cd71152da73e3e41c1
Top:    fce4c904c269311e8b7e1771a13ffd8aa4ae905f
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-13 11:44:15 +0800

Refresh of xdp1_user-support-ifindex-for

---

diff --git a/samples/bpf/xdp1_user.c b/samples/bpf/xdp1_user.c
index caf3f7d..fcb413d 100644
--- a/samples/bpf/xdp1_user.c
+++ b/samples/bpf/xdp1_user.c
@@ -75,12 +75,11 @@ int main(int argc, char **argv)
 	struct bpf_prog_load_attr prog_load_attr = {
 		.prog_type	= BPF_PROG_TYPE_XDP,
 	};
-	const char *optstr = "SND";
+	const char *optstr = "SNO";
 	int prog_fd, map_fd, opt;
 	struct bpf_object *obj;
 	struct bpf_map *map;
 	char filename[256];
-	int ifindex = -1;
 
 	while ((opt = getopt(argc, argv, optstr)) != -1) {
 		switch (opt) {
@@ -90,8 +89,8 @@ int main(int argc, char **argv)
 		case 'N':
 			xdp_flags |= XDP_FLAGS_DRV_MODE;
 			break;
-		case 'D':
-			ifindex = atoi(optarg);
+		case 'O':
+			xdp_flags |= XDP_FLAGS_HW_MODE;
 			break;
 		default:
 			usage(basename(argv[0]));
@@ -113,7 +112,7 @@ int main(int argc, char **argv)
 
 	snprintf(filename, sizeof(filename), "%s_kern.o", argv[0]);
 	prog_load_attr.file = filename;
-	if (ifindex != -1)
+	if (xdp_flags & XDP_FLAGS_HW_MODE)
 		prog_load_attr.ifindex = ifindex;
 
 	if (bpf_prog_load_xattr(&prog_load_attr, &obj, &prog_fd))
